app/bin/databricks_common_utils.py

```diff
@@ -255,66 +259,151 @@ def format_to_json_parameters(params):
+    data_dict = {"grant_type": "client_credentials", "scope": const.SCOPE}
+
+    data_dict["client_id"] = aad_client_id
+    data_dict["client_secret"] = aad_client_secret
     data_encoded = urlencode(data_dict)
+
+    if proxy_settings:
+        proxy_settings.pop("use_for_oauth")
+
     while retry:
         try:
-            resp = requests.post(token_url, headers=headers, data=data_encoded,
-                                 proxies=proxy_settings, verify=const.VERIFY_SSL)
+            resp = requests.post(
+                token_url,
+                headers=headers,
+                data=data_encoded,
+                proxies=proxy_settings,
+                verify=const.VERIFY_SSL,
+                timeout=const.TIMEOUT
+            )
             resp.raise_for_status()
             response = resp.json()
-            access_token = response.get("access_token")
-            if not all([tenant_id, client_id, client_secret]):
-                save_databricks_aad_access_token(session_key, access_token, client_sec)
-            return access_token
+            aad_access_token = response.get("access_token")
+            if not all([aad_tenant_id, aad_client_id, aad_client_secret]):
+                save_databricks_aad_access_token(
+                    account_name, session_key, aad_access_token, aad_client_secret
+                )
+            return aad_access_token
         except Exception as e:
             retry -= 1
-            error_code = resp.json().get('error_codes')
-            if error_code:
-                error_code = str(error_code[0])
-            if error_code == '700016':
-                msg = 'Invalid Client ID provided.'
-            elif error_code == '90002':
-                msg = 'Invalid Tenant ID provided.'
-            elif error_code == '7000215':
-                msg = 'Invalid Client Secret provided.'
-            elif resp.status_code == 403:
-                msg = "Client secret may have expired. Please configure a valid Client secret."
-            elif resp.status_code == 404:
-                msg = "Invalid API endpoint."
-            elif resp.status_code == 500:
-                msg = "Internal server error."
-            elif resp.status_code == 400:
-                msg = resp.json().get("message", "Bad request. The request is malformed.")
-            elif resp.status_code == 429:
-                msg = "API limit exceeded. Please try again after some time.",
-            elif resp.status_code not in (200, 201):
-                msg = 'Response status: {}. Unable to validate Azure Active Directory Credentials.'\
-                    'Check logs for more details.'.format(str(resp.status_code))
+            if "resp" in locals():
+                error_code = resp.json().get("error_codes")
+                if error_code:
+                    error_code = str(error_code[0])
+                if error_code in list(const.ERROR_CODE.keys()):
```

Comment:
if `const.ERROR_CODE` is a map, then it's just enough: `if error_code in const.ERROR_CODE`